File: java/arcade/PalindromeRearranging.java
public class PalindromeRearranging {

	boolean palindromeRearranging(String inputString) {
		int[] abc = new int[26];
		int c = 0;
		for(int i = 0; i < inputString.length(); i++) {
		    abc[inputString.charAt(i) - 'a']++;
		}
		for(int i : abc) { c += (i % 2 == 1) ? 1 : 0; }
		return (inputString.length() % 2 == c);
	}
}

File: ruby/arcade/palindromeRearranging.rb
def palindromeRearranging(inputString)
	a, c = Array.new(26, 0), 0
    (0...inputString.size).each { |i| a[inputString[i].ord - "a".ord] += 1 }
    a.each { |i| c += 1 if i % 2 == 1 }
    return inputString.size % 2 == c
end